Month3/Project1/Project1/ArrayVectors.cpp

```diff
@@ -0,0 +1,280 @@
+	{
+		std::cin.clear();
+		std::cin.ignore(100, '\n');
+		std::cout << "Invalid input,try again" << std::endl;
+		std::cin >> Input;
+	}
+	return Input;
+}
+
+
+
+
+
+float FCircle::GetPerimeter() const
+{
+	return (float)M_PI * (float)pow(Radius, 2);
+}
+
+float FCircle::GetArea() const
+{
+	const float Area = (2 * M_PI) * Radius;
+	return  Area;
+}
+
+
+
+
+float FSquare::GetArea() const
+{
+	return (float)pow(SideLenght, 2);
+}
+float FSquare::GetPerimeter() const
+{
+	return SideLenght * 4;
+}
+
+
+
+
+void BuiltInArray()
+{
+	std::cout << "How many shapes you want to create" << std::endl;
+	int ShapesNumber = ValidateIntInput();
+	FShape** ArrShapes = new FShape * [ShapesNumber];
+	int Position;
+	int Shape;
+	float Radius;
+	float SideLeght;
+
+	for (int CurrentIndex = 0; CurrentIndex < ShapesNumber; CurrentIndex++)
+	{
+		Position = CurrentIndex + 1;
+		std::cout << "What shape are you looking for your position number " << Position << std::endl;
+		std::cout << "1. Circle" << std::endl;
+		std::cout << "2. Square" << std::endl;
+		Shape = ValidateIntInput();
+
+		while (Shape > 2)
+		{
+			std::cin.clear();
+			std::cin.ignore(100, '\n');
+			std::cout << "Sorry, You only can Select Cicle or Square, try again" << std::endl;
+			Shape = ValidateIntInput();
+		}
+
+		if (Shape == 1)
+		{
+			std::cout << "you select circle, now please enter a radius for the circle" << std::endl;
+			Radius = ValidateFloatInput();
+			ArrShapes[CurrentIndex] = new FCircle(Radius);
+		}
+		if (Shape == 2)
+		{
+			std::cout << "you select square, now please enter side lenght" << std::endl;
+			SideLeght = ValidateFloatInput();
+			ArrShapes[CurrentIndex] = new FSquare(SideLeght);
+		}
+
+	}
+	system("CLS");
+	for (int CurrentIndex = 0; CurrentIndex < ShapesNumber; CurrentIndex++)
+	{
+		std::cout << "The Area of your shape is: " << ArrShapes[CurrentIndex]->GetArea() << std::endl;
+		std::cout << "The Perimeter of your shape is: " << ArrShapes[CurrentIndex]->GetPerimeter() << std::endl;
+		std::cout << std::endl;
+	}
+	
+	delete[] ArrShapes;
+}
+
+
+void DinamicArray() 
+{
+	TDinamicArray <FShape*> ArrShapes(1);
+	std::cout << "What do you want to do" << std::endl;
+	std::cout << "1. Add a Shape" << std::endl;
+	std::cout << "2. You want to print the shapes you already have?" << std::endl;
+	int Choice = ValidateIntInput();
+	while (Choice > 2 || Choice < 1) 
+	{
+		std::cin.clear();
+		std::cin.ignore(100, '\n');
+		std::cout << "invalid Input, you only can choice 1 or 2 option, Try Again" << std::endl;
+		Choice = ValidateIntInput();
+	}
+	float Radius;
+	float SideLeght;
+	if (Choice == 1)
+	{
+		std::cout << "you select circle, now please enter a radius for the circle" << std::endl;
+		Radius = ValidateFloatInput();
+		ArrShapes.PushBack(new FCircle(Radius));
+	}
+
+	if (Choice == 2)
+	{
+		std::cout << "you select square, now please enter side lenght" << std::endl;
+		SideLeght = ValidateFloatInput();
+		ArrShapes.PushBack(new FSquare(SideLeght));
+	}
+	
```

Comment:
Something that I did for my project (though also not necessary) is to have a function `FShape* CreateShape()`. Notice that you're doing the same thing for all arrays, and both FCircle and FSquare can be returned as FShape*. This way, you can avoid having to rewrite this for all the array functions, and have the differences (like PushBack for this one) after the function call.